pkg/cvo/sync_worker.go

```diff
@@ -416,20 +418,10 @@ func (w *SyncWorker) calculateNext(work *SyncWork) bool {
 	// the state Update() calculated (to allow us to start in reconciling)
 	if work.Empty() {
 		work.State = w.work.State
+		work.Attempt = 0
 	} else if changed {
 		work.State = payload.UpdatingPayload
-	}
-	// always clear the completed variable if we are not reconciling
-	if work.State != payload.ReconcilingPayload {
-		work.Completed = 0
-	}
-
-	// track how many times we have tried the current payload in the current
-	// state
-	if changed || w.work.State != work.State {
 		work.Attempt = 0
```

Comment:
It is not clear why we need to make Attempt= 0 here?  Because it means `equalSyncWork returned false`

Reply:
We want to reset the count, because we are about to begin our first attempt at the new target.